File: examples/clients/python/http_client.py
#!/usr/bin/env python3
import json
import os
import urllib.request


BASE_URL = os.environ.get("LEASH_URL", "http://127.0.0.1:8000").rstrip("/")


def request_json(method, path):
    request = urllib.request.Request(f"{BASE_URL}{path}", method=method)
    if method == "POST":
        request.data = b""
    with urllib.request.urlopen(request, timeout=5) as response:
        return json.loads(response.read().decode("utf-8"))


def main():
    health = request_json("GET", "/health")
    telemetry = request_json("GET", "/telemetry")
    stop = request_json("POST", "/stop")

    if health.get("ok") is not True:
        raise SystemExit("health did not report ok=true")
    if not telemetry.get("robot") or telemetry.get("profile") != "sim":
        raise SystemExit("telemetry did not look like a sim frame")
    if stop.get("ok") is not True:
        raise SystemExit("stop did not report ok=true")

    print(json.dumps({
        "ok": True,
        "runtime": "python",
        "profile": health["profile"],
        "robot": telemetry["robot"],
        "stop": stop,
    }, sort_keys=True))


if __name__ == "__main__":
    main()

File: examples/clients/node/http-client.mjs
const baseUrl = (process.env.LEASH_URL || "http://127.0.0.1:8000").replace(/\/$/, "");

async function requestJson(method, path) {
  const response = await fetch(`${baseUrl}${path}`, { method });
  if (!response.ok) {
    throw new Error(`${method} ${path} returned HTTP ${response.status}`);
  }
  return response.json();
}

const health = await requestJson("GET", "/health");
const telemetry = await requestJson("GET", "/telemetry");
const stop = await requestJson("POST", "/stop");

if (health.ok !== true) {
  throw new Error("health did not report ok=true");
}
if (!telemetry.robot || telemetry.profile !== "sim") {
  throw new Error("telemetry did not look like a sim frame");
}
if (stop.ok !== true) {
  throw new Error("stop did not report ok=true");
}

console.log(JSON.stringify({
  ok: true,
  runtime: "node",
  profile: health.profile,
  robot: telemetry.robot,
  stop,
}));
